rockcraft.yaml

```diff
@@ -87,9 +89,51 @@ parts:
     stage:
       - opt/spark/jars
 
-  user-setup:
+  client-scripts:
     plugin: nil
     after: [hadoop-jars]
+    build-packages:
+      - wget
+    overlay-packages:
+      - python3-pip
+    overlay-script: |
+      SPARK_CLIENT_PKG_VERSION='0.0.2'
+      mkdir -p $CRAFT_PART_INSTALL/opt/spark/python/dist
+      cd $CRAFT_PART_INSTALL/opt/spark/python/dist
+      echo "Downloading latest available Spark wheel package release v${SPARK_CLIENT_PKG_VERSION}."
+      wget https://github.com/canonical/spark-client-snap/releases/download/v${SPARK_CLIENT_PKG_VERSION}/spark_client-${SPARK_CLIENT_PKG_VERSION}-py3-none-any.whl --no-check-certificate
+      if  [[ $? -ne 0 ]]
+        then
+          echo "DOWNLOAD ERROR: Spark client package v${SPARK_CLIENT_PKG_VERSION} could not be downloaded properly! Exiting...."
+          exit 1
+      fi
+      pip install --target=${CRAFT_PART_INSTALL}/opt/spark/python/dist ./spark_client-${SPARK_CLIENT_PKG_VERSION}-py3-none-any.whl
+    stage:
+      - opt/spark/python/dist
+
+  kubectl-bin:
+    plugin: nil
+    after: [client-scripts]
+    build-packages:
+      - curl
+    overlay-script: |
+      mkdir -p $CRAFT_PART_INSTALL/opt/spark
+      cd $CRAFT_PART_INSTALL/opt/spark
+      curl -LO -s "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+      curl -LO "https://dl.k8s.io/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl.sha256"
+      echo "$(cat kubectl.sha256)  kubectl" | sha256sum --check
```

Comment:
Curl in Curl approach will fail in a moment of updating https://dl.k8s.io/release/stable.txt.
I would fetch it first / once and use as variable during fetching binary and checksum.

Reply:
good point. done.